### DISCUSSION TITLE
How to customize drag and drop handling?

### DISCUSSION THREAD

**Discussion by Kebab11noel** [mdx-editor/editor#36] [Q&A] [answered]
In my case i have a row for image previews, however when i drag them over to the editor they get embedded the same as the original element, but i'd like to convert them to an admonition like `::img[alt text here]{image identifier here}`

**Comment by petyosi** [comment, marked as answer]
Hey @Kebab11noel, 

Thank you for giving the editor a try! Based on what you're saying, I assume that you would rather store images as markdown directives and not as html tags (or markdown images). 

To do so, you don't need to customize the drag and drop handling, but rather tap into the markdown processing itself and swap the way images are handled. [Here's an example of a plugin that does that](https://codesandbox.io/s/image-as-directive-d3hkl5?file=/src/App.tsx). Notice however, that the images are inline elements. So I am using the `TextDirective` `:img` rather than the leaf (block) `::img`. 

Let me know if I got that right or if you need any further help.

**Reply by Kebab11noel** [reply]
Thank you for the example, it helped me to get a better idea how this works, however in my JS project i can't get it to work, in the console i get: `index.js:113 Uncaught Error: Cannot handle unknown node textDirective`.

Also a little more explanation why i need this: I have an image cdn that provides different sizes of the image so from `:img...` I'd like to generate a `picture` tag with sources for different screen sizes, so the URL also needs to be assembled from `BASE_URL<image_identifier>/<variant>`, but I can probably solve this part since in the editor I'm good with the single variant i.e. the simple img tag.

**Comment by petyosi** [comment]
The error suggests that the mdast visitor has not caught the parsed MDAST node - or, maybe, you have another directive node that needs parsing? That's just a guess, of course. You can do some debugging at this line and [see what happens there](https://codesandbox.io/s/image-as-directive-d3hkl5?file=/src/App.tsx:689-764). 

If you can reproduce the problem somewhere in sample, I will take a look.

**Reply by Kebab11noel** [reply]
Well, it might have been a dependency issue, I tried to create a reproducer which worked, so i just took the dependencies and pasted over the existing and then it worked, but sadly i realized this later so I don't have the original versions I tried.
